pacman/model/graphs/machine/constant_sdram_machine_partition.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2019-2020 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from pacman.model.graphs.machine import AbstractSDRAMPartition
+from pacman.model.graphs import AbstractSingleSourcePartition
+from spinn_utilities.overrides import overrides
+from pacman.exceptions import (
+    PacmanConfigurationException, PartitionMissingEdgesException,
+    SDRAMEdgeSizeException)
+from pacman.model.graphs.common import EdgeTrafficType
+from pacman.model.graphs.machine import SDRAMMachineEdge
+
+
+class ConstantSDRAMMachinePartition(
+        AbstractSingleSourcePartition, AbstractSDRAMPartition):
+
+    __slots__ = [
+        # The sdram base address for this partition.
+        "_sdram_base_address",
+        # The sdram size of every edge or None if no edge added.
+        "_sdram_size",
+    ]
+
+    MISSING_EDGE_ERROR_MESSAGE = "Partition {} has no edges"
+
+    def __init__(self, identifier, pre_vertex, label):
+        super(ConstantSDRAMMachinePartition, self).__init__(
+            pre_vertex, identifier, allowed_edge_types=SDRAMMachineEdge,
+            constraints=None, label=label, traffic_weight=1,
+            class_name="ConstantSdramMachinePartition")
+        self._sdram_size = None
+        self._sdram_base_address = None
+
+    @property
+    @overrides(AbstractSDRAMPartition.traffic_type)
+    def traffic_type(self):
+        return EdgeTrafficType.SDRAM
+
+    @overrides(AbstractSingleSourcePartition.add_edge)
+    def add_edge(self, edge, graph_code):
+        if self._sdram_size is None:
+            self._sdram_size = edge.sdram_size
+        elif self._sdram_size != edge.sdram_size:
+            raise SDRAMEdgeSizeException(
+                "The edges within the constant sdram partition {} have "
```

Comment:
Missing value to put in the exception message? (Think it might be a "self"?)

Reply:
fixed